app/classes/query/modules/initialization.rb

```diff
@@ -34,10 +34,16 @@ def escape(val)
       #   @where << "names.id IN (#{set})"
       #
       def clean_id_set(ids)
-        set = ids.map(&:to_i).uniq[0, MO.query_max_array].map(&:to_s).join(",")
+        set = limited_id_set(ids).map(&:to_s).join(",")
         set.presence || "-1"
       end
 
+      # array of max of 1000 unique ids for use with Arel "in"
+      #    where(<x>.in(limited_id_set(ids)))
+      def limited_id_set(ids)
+        ids.map(&:to_i).uniq[0, MO.query_max_array]
```

Comment:
For reference or those who are curious:  ActiveRecord does handle the empty set reasonably well.  It converts that partial condition to just "1 = 0".  Nice.